updater/internal/install/install.go

```diff
@@ -223,3 +243,216 @@ func skipConfigFiles(path string) bool {
+	var endpoint, secretKey string
+	if endpoint, ok = manager["endpoint"].(string); !ok {
+		return errors.New("read in endpoint")
+	}
+	if secretKey, ok = manager["secret_key"].(string); !ok {
+		return errors.New("read in secret key")
+	}
+
+	// Construct new supervisor config
+	cfg := &SupervisorConfig{
+		Server: Server{
+			Endpoint: endpoint,
+			Headers: Headers{
+				Authorization: "Secret-Key " + secretKey,
+				AgentIDFormat: "ULID",
+			},
+			TLS: TLS{
+				Insecure:           true,
+				InsecureSkipVerify: true,
+			},
+		},
+		Capabilities: Capabilities{
+			AcceptsRemoteConfig: true,
+			ReportsRemoteConfig: true,
+		},
+		Agent: Agent{
+			Executable:      filepath.Join(installDir, "observiq-otel-collector"),
+			HealthCheckPort: hcePort,
+		},
+		Storage: Storage{
+			Directory: filepath.Join(installDir, "supervisor_storage"),
+		},
+	}
+
+	// write cfg to file
+	err := writeSupervisorConfig(logger, cfg, installDir, rb)
+	if err != nil {
+		return fmt.Errorf("write supervisor config")
+	}
+	return nil
+}
+
+func writeSupervisorConfig(logger *zap.Logger, supervisorCfg *SupervisorConfig, installDir string, rb rollback.Rollbacker) error {
+	supervisorYaml, err := yaml.Marshal(supervisorCfg)
+	if err != nil {
+		return fmt.Errorf("marshal supervisor yaml: %w", err)
+	}
+
+	supervisorPath := filepath.Join(installDir, "supervisor.yaml")
+
+	// We create the action record here, because we want to record the file does not exist
+	// before we open the file (which will end up creating the file).
+	// Use CopyFileAction because rollback can be used to delete supervisor.yaml.
+	// Leave fromPathRel & backupDir empty bc supervisor is a new file.
+	cfa, err := action.NewCopyFileAction(logger, "", supervisorPath, "")
+	if err != nil {
+		return fmt.Errorf("create supervisor.yaml copy file action: %w", err)
+	}
+	rb.AppendAction(cfa)
+
+	supervisorFile, err := os.OpenFile(filepath.Clean(supervisorPath), os.O_CREATE|os.O_WRONLY, fs.FileMode(0600))
+	if err != nil {
+		return fmt.Errorf("open supervisor config file: %w", err)
+	}
+	defer func() {
+		err := supervisorFile.Close()
+		if err != nil {
+			logger.Error("close supervisor config file", zap.Error(err))
+		}
+	}()
+
+	_, err = supervisorFile.Write(supervisorYaml)
+	if err != nil {
+		return fmt.Errorf("write supervisor config: %w", err)
+	}
+	return nil
+}
+
+// PersistentState stores the ID of the collector
+type PersistentState struct {
+	InstanceID string `yaml:"instance_id"`
+}
+
+func handleAgentIDConversion(logger *zap.Logger, manager map[string]any, installDir string, rb rollback.Rollbacker) error {
+	// Retrieve agent id and convert to UUID
+	agentID, ok := manager["agent_id"].(string)
+	if !ok {
+		return fmt.Errorf("read in agent id")
+	}
+	agentULID, err := ulid.Parse(agentID)
+	if err != nil {
+		return fmt.Errorf("parse agent id into a ULID: %w", err)
+	}
+	ulidBytes, err := agentULID.MarshalBinary()
+	if err != nil {
+		return fmt.Errorf("marshal ulid to bytes: %w", err)
+	}
```

Comment:
With #1907 merged, we'll want to check if it's a UUID as well.

Side note, it does suck a bit that this will cause the agent to report itself with a new ID. I don't think there's a way around it though.